File: app/src/main/java/soup/movie/MovieApplication.java
package soup.movie;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;
import soup.movie.di.DaggerApplicationComponent;

public class MovieApplication extends DaggerApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        BuildType.init(this);
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerApplicationComponent.builder().application(this).build();
    }
}

File: app/src/main/java/soup/movie/MovieApplication.kt
package soup.movie

import android.app.Application
import android.content.Context
import com.google.android.play.core.splitcompat.SplitCompat
import com.jakewharton.threetenabp.AndroidThreeTen
import dagger.hilt.android.HiltAndroidApp
import soup.movie.notification.NotificationChannels
import soup.movie.theme.ThemeOptionManager
import javax.inject.Inject

@HiltAndroidApp
class MovieApplication : Application() {

    @Inject lateinit var themeOptionManager: ThemeOptionManager

    override fun onCreate() {
        super.onCreate()
        AndroidThreeTen.init(this)
        NotificationChannels.initialize(this)
        themeOptionManager.initialize()
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        SplitCompat.install(this)
    }
}
